src/main/scala/overflowdb/codegen/CodeGen.scala

```diff
@@ -716,78 +897,112 @@ def writeConstants(outputDir: JFile): JFile = {
          |/** base type for all nodes that can be added to a graph, e.g. the diffgraph */
          |trait NewNode extends CpgNode {
          |  def properties: Map[String, Any]
-         |  def containedNodesByLocalName: Map[String, List[${DefaultNodeTypes.NodeClassname}]]
-         |  def allContainedNodes: List[${DefaultNodeTypes.NodeClassname}] = containedNodesByLocalName.values.flatten.toList
          |}
          |""".stripMargin
 
     def generateNewNodeSource(nodeType: NodeType, keys: List[Property]) = {
-      val fields: String = {
-        val forKeys = keys.map { key =>
-          val optionalDefault =
-            if (getHigherType(key) == HigherValueType.Option) " = None"
-            else if (key.valueType == "int") " = -1"
-            else if (getHigherType(key) == HigherValueType.None && key.valueType == "string") """ ="" """
-	    else if (getHigherType(key) == HigherValueType.None && key.valueType == "boolean") """ =false """
-            else if (getHigherType(key) == HigherValueType.List) "= List()"
-            else ""
-          s"${camelCase(key.name)}: ${getCompleteType(key)} $optionalDefault"
+      var fieldsM = List[(String, String, Option[String])]()
```

Comment:
naming.. what does the `M` in `fieldsM` stand for? for readability it may be useful to either create type aliases, e.g. `type OptionalDefault  = Option[String]`, or use a `case class(name: String, typ: String, defaultMaybe: Option[String)` for readability and type safety